server/src/sim.rs

```diff
@@ -109,6 +112,104 @@ impl Sim {
+    ) -> anyhow::Result<()> {
+        let entity_id = EntityId::from_bits(u64::from_le_bytes(save_entity.entity));
+        let mut entity_builder = EntityBuilder::new();
+        entity_builder.add(entity_id);
+        for (component_type, component_bytes) in save_entity.components {
+            self.load_component(
+                read,
+                &mut entity_builder,
+                node,
+                ComponentType::try_from(component_type as i32).unwrap(),
+                component_bytes,
+            )?;
+        }
+        let entity = self.world.spawn(entity_builder.build());
+        self.graph_entities.insert(node, entity);
+        self.entity_ids.insert(entity_id, entity);
+        Ok(())
+    }
+
+    fn load_component(
+        &mut self,
+        read: &mut save::Reader,
+        entity_builder: &mut EntityBuilder,
+        node: NodeId,
+        component_type: ComponentType,
+        component_bytes: Vec<u8>,
+    ) -> anyhow::Result<()> {
+        match component_type {
+            ComponentType::Position => {
+                let column_slice: [f32; 16] = postcard::from_bytes(&component_bytes)?;
+                entity_builder.add(Position {
+                    node,
+                    local: na::Matrix4::from_column_slice(&column_slice),
+                });
+            }
+            ComponentType::Name => {
+                let name = String::from_utf8(component_bytes)?;
+                // Ensure that every node occupied by a character is generated.
+                if let Some(character) = read.get_character(&name)? {
+                    let mut current_node = NodeId::ROOT;
+                    for side in character
+                        .path
+                        .into_iter()
+                        .map(|side| Side::from_index(side as usize))
+                    {
+                        current_node = self.graph.ensure_neighbor(current_node, side);
+                    }
+                    if current_node != node {
+                        // Skip loading named entities that are in the wrong place. This can happen
+                        // when there are multiple entities with the same name, which has been possible
+                        // in the past.
+                        return Ok(());
+                    }
+                } else {
+                    // Skip loading named entities that lack path information.
+                    return Ok(());
+                }
+                // Prepare all relevant components that are needed to support ComponentType::Name
+                entity_builder.add(InactiveCharacter(Character {
+                    name,
+                    state: CharacterState {
+                        velocity: na::Vector3::zeros(),
+                        on_ground: false,
+                        orientation: na::UnitQuaternion::identity(),
+                    },
+                }));
+                entity_builder.add(CharacterInput {
```

Comment:
Why do inactive characters need input state? Should this be inserted during activation instead?

Reply:
That would be a good idea. I made the change.